### PR TITLE
Add `--dev` and `--debug` flag to start command

### PR DESCRIPTION
**Pull request by cdupuis**

### CODE REVIEW

File: lib/spawn.ts [hunk 1]
```diff
@@ -64,7 +64,7 @@ export async function spawnBinary(opts: SpawnOptions): Promise<number> {
 export async function spawnJs(opts: SpawnOptions): Promise<number> {
     opts.cwd = path.resolve(opts.cwd);
     const script = path.join(opts.cwd, "node_modules", "@atomist", "automation-client", opts.command);
-    opts.args = [script, ...opts.args];
+    opts.args = [...opts.args, script];
```
Comment (cdupuis) [inline lib/spawn.ts]: This method wasn't called from any other place, so it seemed ok to change it.

File: lib/start.ts [hunk 2]
```diff
@@ -39,10 +46,19 @@ export async function start(opts: StartOptions): Promise<number> {
...
+        };
+        return spawnBinary(spawnOpts);
+    } else {
+        const spawnOpts = {
+            ...opts,
+            command: path.join("bin", opts.dev ? "start-dev.js" : "start.js"),
+            args: ["--inspect"],
+        };
+        return spawnJs(spawnOpts);
+    }
```
Comment (cdupuis) [inline lib/start.ts]: I'm unsure on this. I needed a way to pass an additional parameter to `node` and I couldn't find a way to do that when running the nom scripts. Maybe you do.

File: lib/spawn.ts (hunk 1 above)
Comment (ddgenome) [inline lib/spawn.ts]: Won't this break things that currently rely on the fact that the args are passed to the script, not `node`?

File: lib/start.ts [hunk 3]
```diff
@@ -39,10 +46,19 @@ export async function start(opts: StartOptions): Promise<number> {
...
+        const spawnOpts = {
+            ...opts,
+            command: opts.dev ? "atm-start-dev" : "atm-start",
+            args: [] as string[],
+        };
+        return spawnBinary(spawnOpts);
+    } else {
+        const spawnOpts = {
+            ...opts,
+            command: path.join("bin", opts.dev ? "start-dev.js" : "start.js"),
```
Comment (ddgenome) [inline lib/start.ts]: Should `--debug` imply `--dev`?

File: lib/spawn.ts (hunk 1 above)
Comment (cdupuis) [reply lib/spawn.ts]: This function wasn't used anywhere in the CLI code base, so that felt ok to change.

File: lib/start.ts (hunk 3 above)
Comment (cdupuis) [reply lib/start.ts]: I tried that, but didn't like it.

**Comment by cdupuis** [comment]
@ddgenome instead of `--dev` should this be `--reload`?

**Comment by ddgenome** [comment]
`--watch` is nice

**Comment by cdupuis** [comment]
> `--watch` is nice

With `--watch` I think we can keep `--debug` because it is different, ok?

File: lib/spawn.ts (hunk 1 above)
Comment (ddgenome) [reply lib/spawn.ts]: Huh, I guess all the other scripts that used that were moved to automation-client.

File: lib/start.ts (hunk 2 above)
Comment (ddgenome) [reply lib/start.ts]: I guess since this is not used, it's no big deal. If it were me, I'd create another property. Something like `nodeArgs`.

Comment (cdupuis) [reply lib/start.ts]: Makes sense, I'll do that.

File: lib/start.ts (hunk 3 above)
Comment (ddgenome) [reply lib/start.ts]: I was trying to ask about the other way around. If you want to run the debugger, should you also always enable watching?

**Comment by ddgenome** [comment]
As you wish.

**Pull request merged**